### PR TITLE
ENH add mockmpi and parallel_statistics

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge/staged-recipes This one is ready for review!

**Comment by beckermr** [comment]
This one is noarch so only linux matters.

**Comment by ocefpaf** [comment]
> No we don't do that for noarch. Noarch only builds on linux but can run on any platform where it is installable.

So you do expect it to work on Windows. My understanding of your comment above is that it would not. I did that to a few to have "semi-arch" on Python version but prevent the installation on Windows. It works great.

**Comment by beckermr** [comment]
I am confused. This package is noarch. It builds on linux and will work on any platform where it can be installed. We've been things this way for years now.

The windows failure is unrelated

```

WARNING:conda_build.config:Setting build platform. This is only useful when pretending to be on another platform, such as for rendering necessary dependencies on a non-native platform. I trust that you know what you're doing.
WARNING:D:\a\1\s\.ci_support\compute_build_graph.py:Dependency protozfits, version =2.0 is not installable from your channels: ['local', 'conda-forge', 'conda-forge'] with subdir win-64.  Seeing if we can build it...
rendering D:\a\1\s\recipes\litex for win-64
rendering D:\a\1\s\recipes\ml_dtypes for win-64
WARNING:D:\a\1\s\.ci_support\compute_build_graph.py:Dependency pyink, version any is not installable from your channels: ['local', 'conda-forge', 'conda-forge'] with subdir win-64.  Seeing if we can build it...
rendering D:\a\1\s\recipes\pyink for win-64
Traceback (most recent call last):
  File "D:\a\1\s\.ci_support\build_all.py", line 218, in <module>
    build_all(os.path.join(root_dir, "recipes"), args.arch)
  File "D:\a\1\s\.ci_support\build_all.py", line 118, in build_all
    build_folders(recipes_dir, folders, arch, channel_urls)
  File "D:\a\1\s\.ci_support\build_all.py", line 155, in build_folders
    G = construct_graph(recipes_dir, worker=worker, run='build',
  File "D:\a\1\s\.ci_support\compute_build_graph.py", line 401, in construct_graph
    add_recipe_to_graph(recipe_dir, graph, run, worker, conda_resolve,
  File "D:\a\1\s\.ci_support\compute_build_graph.py", line 233, in add_recipe_to_graph
    add_dependency_nodes_and_edges(name, graph, run, worker, conda_resolve, config=config,
  File "D:\a\1\s\.ci_support\compute_build_graph.py", line 466, in add_dependency_nodes_and_edges
    recipe_dir = _buildable(dep, version, recipes_dir, worker, metadata.config,
  File "D:\a\1\s\.ci_support\compute_build_graph.py", line 442, in _buildable
    available = match_peer_job(ms, m)
  File "D:\a\1\s\.ci_support\compute_build_graph.py", line 256, in match_peer_job
    if conda_interface.conda_43:
AttributeError: module 'conda_build.conda_interface' has no attribute 'conda_43'
```

Looks like conda_build changed an API.

**Comment by ocefpaf** [comment]
> I am confused. This package is noarch. It builds on linux and will work on any platform where it can be installed. We've been doing things this way for years now.

Matt, I was confused by your comment that Windows didn't matter. If it does work on Windows and the failure here doesn't matter, then merging is fine. But if this was a Linux only pkg, like you implied above, then restricting it to installing on Linux would make sense.

TL;DR we **can** have noarch that installs only on Linux, Windows, or Mac for a while now. Say, a Python noarch that relies on a Linux only CLI command for example.

**Comment by beckermr** [comment]
Ahhhhh ok. I changed my comment above to clarify. My point was simply that only linux needs to pass, and indeed this package will run fine on windows, osx or linux.
